pkg/registry/apis/provisioning/list.go

```diff
@@ -0,0 +1,72 @@
+package provisioning
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	provisioning "github.com/grafana/grafana/pkg/apis/provisioning/v0alpha1"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/resources"
+)
+
+type listConnector struct {
+	getter RepoGetter
+	lister resources.ObjectLister
+}
+
+func (*listConnector) New() runtime.Object {
+	return &provisioning.TestResults{}
+}
+
+func (*listConnector) Destroy() {}
+
+func (*listConnector) ProducesMIMETypes(verb string) []string {
+	return []string{"application/json"}
+}
+
+func (*listConnector) ProducesObject(verb string) any {
+	return &provisioning.TestResults{}
+}
+
+func (*listConnector) ConnectMethods() []string {
+	return []string{http.MethodGet}
+}
+
+func (*listConnector) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, false, ""
+}
+
+func (s *listConnector) Connect(ctx context.Context, name string, opts runtime.Object, responder rest.Responder) (http.Handler, error) {
+	ns, ok := request.NamespaceFrom(ctx)
+	if !ok {
+		return nil, fmt.Errorf("missing namespace")
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Query().Has("count") {
```

Comment:
if we end up using boolean flag, we should check if it's true and not only its existence. Otherwise, you have to trust that the caller in implemented it in a way that it doesn't add the query param at all. They could implement it using a `false` as default and always passing it.

Reply:
removed -- and added an endpoint at the root